src/transaction/components/TransactionProvider.tsx

```diff
@@ -122,16 +198,13 @@ export function TransactionProvider({
 
   // Set transaction pending status when writeContracts or writeContract is pending
   useEffect(() => {
-    if (
-      statusWriteContracts === 'pending' ||
-      statusWriteContract === 'pending'
-    ) {
+    if (transactionStatus === 'pending') {
       setLifeCycleStatus({
         statusName: 'transactionPending',
         statusData: null,
```

Comment:
Open question should any particular `statusData` being provided here?

Reply:
I was toying around with adding the `transactions` variable here, however didn't want the PR to get too big.

we can add 2 tasks as follow-ups:
- fix the `pending` state inbetween multiple transactions for EOAs
- expose each `transaction` as status data in `transactionPending` and `success`

Reply:
I like that as follow up PRs. Yeah we can merge this one as it is, and do more polish after.